Tribler/Core/CreditMining/BoostingSource.py

```diff
@@ -259,11 +259,11 @@ def _update(self):
 
             # it's highly probable the checktor function is running at this time (if it's already running)
             # if not running, start the checker
-
-            task_call = self.register_task(hexlify(self.source) + "_checktor", LoopingCall(self._check_tor))
-            if task_call:
-                self._logger.debug("Registering check torrent function")
-                task_call.start(self.check_torrent_interval, now=True)
+            if not self.is_pending_task_active(hexlify(self.source) + "_checktor"):
```

Comment:
Is it expected that this method is called again when the task is possibly still pending? If so, please add this to the comment above the line. Otherwise, it might look like a workaround.